drivers/include/periph/i2c.h

```diff
@@ -272,9 +240,13 @@ int i2c_release(i2c_t dev);
  * @param[out] data         memory location to store received data
  * @param[in]  flags        optional flags (see @ref i2c_flags_t)
  *
- * @return                  I2C_ACK on successful transfer of @p len byte
- * @return                  I2C_ADDR_NACK if response to address byte was NACK
- * @return                  I2C_ERR for any other error
+ * @return                  0 Success
+ * @return                  -EIO Slave device doesn't ACK the byte
```

Comment:
maybe add a `when` between error code and the rest. `-EIO when slave device doesn't ACK the byte`.